tests/conftest.py

```diff
@@ -1730,17 +1732,81 @@ def duthost_console(duthosts, enum_supervisor_dut_hostname, localhost, conn_grap
-                       console_port=console_port,
-                       sonic_username=creds['sonicadmin_user'],
-                       sonic_password=[creds['sonicadmin_password'], sonicadmin_alt_password],
-                       console_username=console_username,
-                       console_password=creds['console_password'][console_type])
+
+    # Set up console host
+    host = None
+    try:
+        host = ConsoleHost(console_type=console_type,
+                           console_host=console_host,
+                           console_port=console_port,
+                           sonic_username=creds['sonicadmin_user'],
+                           sonic_password=[creds['sonicadmin_password'], sonicadmin_alt_password],
+                           console_username=console_username,
+                           console_password=creds['console_password'][console_type])
+    except PortInUseException:
+        # Clear the console port, then attempt setup again
+        duthost_clear_console_port(
+            console_host=console_host,
+            console_port=console_port,
+            console_password=creds['console_password'][console_type]
+        )
+
+        host = ConsoleHost(console_type=console_type,
+                           console_host=console_host,
+                           console_port=console_port,
+                           sonic_username=creds['sonicadmin_user'],
+                           sonic_password=[creds['sonicadmin_password'], sonicadmin_alt_password],
+                           console_username=console_username,
+                           console_password=creds['console_password'][console_type])
+
+    if host is None:
+        raise Exception("DUT console host was not setup correctly.")
+
     yield host
     host.disconnect()
 
 
+def duthost_clear_console_port(console_host, console_port, console_password):
+    """
+    Helper function to clear the console port for a given DUT.
+    Useful when a device has an occupied console port, preventing dut_console tests from running.
+    This function uses the default 'admin' account to log in to the Digi CM 48 config page,
+    so it does not require a username being passed to it.
+
+    Parameters:
+        console_host (str): DUT host's console IP address
+        console_port (str): DUT host's console port, to be cleared
+        console_password (str): Password for the 'admin' account
+    """
+    # Helper function to send a command and optionally wait for pattern
+    def send_command(host, command_str, pattern=""):
+        host.write_channel(command_str + host.RETURN)
+        host.read_until_prompt_or_pattern(pattern=pattern)
+
+    # Command list per the Digi CM 48 configuration page
+    ENTER_SERIAL_PORT_CONFIG = '2'
+    CHOOSE_DUT_CONSOLE_PORT = console_port
+    ENTER_PORT_MANAGEMENT = 'a'
+    RESET_PORT = '1'
+
+    duthost_config_menu = ConsoleHost(
+        console_type=CONSOLE_SSH_DIGI_CONFIG,
```

Comment:
@matthew-soulsby I think most of our console server is not Digi console server, could you please also help with implementation for the type of console server mostly used in lab? 

@yutongzhang-microsoft do you know how to clear occupied console port? Not only for Digi console server, but also for other types?